### PR TITLE
Add notRestoredReasons to PerformanceNavigationTiming attribute

### PR DESCRIPTION
**Pull request by rubberyuzu**
This PR adds notRestoredReasons as a new attribute of PerformanceNavigationTiming.

***
<a href="https://pr-preview.s3.amazonaws.com/rubberyuzu/navigation-timing/pull/195.html" title="Last updated on Mar 4, 2024, 5:49 AM UTC (c506661)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/navigation-timing/195/2aa2a3a...rubberyuzu:c506661.html" title="Last updated on Mar 4, 2024, 5:49 AM UTC (c506661)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -615,6 +622,10 @@ <h2>Creating a navigation timing entry</h2>
         <li>Set |document|'s <span>navigation timing entry</span> to |navigationTimingEntry|.
         <li>Set |navigationTimingEntry|'s <a data-for="PerformanceNavigationTiming">`Critical-CH` restart time</a>
         to |criticalCHRestart|.
+        <li>If |document| is the outermost main frame's document, set |navigationTimingEntry|'s
```
Comment (domenic) [inline index.html]: You can say:

> If _document_'s [=node navigable=] is a [=top-level traversable=]

File: index.html [hunk 2]
```diff
@@ -615,6 +622,10 @@ <h2>Creating a navigation timing entry</h2>
...
+        <a data-for="PerformanceNavigationTiming">not restored reasons</a> to the result of
+        creating a NotRestoredReasons object given |document|'s node navigable's active session
+        history entry's document state's not restored reasons and realm.</li>
```
Comment (domenic) [inline index.html:628]: You can say:

> |document|'s [=node navigable=]'s [=navigable/active session history entry=]'s [=session history entry/document state=]'s [=document state/not restored reasons=] and |document|'s [=relevant realm=].

**Review by domenic — COMMENTED** [review]
Overall I think this PR would be better if you added a small abstraction over in the HTML PR. Something like:

> A document's **not restored reasons** is its node navigable's active session history entry's document state's not restored reasons, if document's node navigable is a top-level traversable; otherwise null.

(Or maybe the "if... otherwise" is not necessary, because you leave it null for those cases? That would be better.)

Then this PR could have only a single dependency across the Performance Timing -> HTML boundary:

> If |document|'s [=Document/not restored reasons=] is not null, then set |navigationTimingEntry|'s [=PerformanceNavigationTiming/not restored reasons=] to the result of [=creating a NotRestoredReasons object=] given |document|'s [=Document/not restored reasons=].

File: index.html (hunk 1 above)
Comment (rubberyuzu) [reply index.html]: Ack. Removed this part because on the HTML spec side, we set iframes' NRR to null.

File: index.html (hunk 2 above)
Comment (rubberyuzu) [reply index.html]: Thanks! done

**Review by domenic — APPROVED** [review]
Looks great!

File: index.html [hunk 3]
```diff
@@ -615,6 +622,10 @@ <h2>Creating a navigation timing entry</h2>
         <li>Set |document|'s <span>navigation timing entry</span> to |navigationTimingEntry|.
         <li>Set |navigationTimingEntry|'s <a data-for="PerformanceNavigationTiming">`Critical-CH` restart time</a>
         to |criticalCHRestart|.
+        <li>If |document|'s [=Document/not restored reasons=] is not null, then set
```
Comment (smaug----) [inline index.html:625]: document's or document's state's? I guess those are the same thing, but oddly enough the HTML spec pr talks about them both.

Comment (smaug----) [reply index.html:625]: Hmm, why do we need the null check here.  Not restored reasons should be initialized with some value. Either with null or valid reasons object.

**Review by smaug---- — APPROVED** [review]

File: index.html (hunk 3 above)
Comment (rubberyuzu) [reply index.html:625]: You're right, I removed the null check.

**Review by noamr — APPROVED** [review]

**Pull request merged**